velox/exec/ContainerRowSerde.h

```diff
@@ -44,6 +44,15 @@ class ContainerRowSerde {
       const Type* type,
       CompareFlags flags);
 
+  /// Return std::nullopt if encountered in NullHandlingMode::StopAtNull mode.
+  /// Support NullHandlingMode::NoStop mode as well and return
+  /// std::optional<int32_t>(0) if null encountered.
```

Comment:
I may have missed this, but where do we return std::nullopt in the leaf cases?

Reply:
I'm not sure I understand your question about ** leaf cases**. 

It will return `std::nullopt`, say when an element of an ArrayVector is null during in StopAtAtNull mode. `BaseVector::compareNulls` returns `std::nullopt` in StopAtAtNull mode when a null is encountered.
```C++
std::optional<int32_t> compareArrays(
    ...
    if (leftNull || rightNull) {
      auto result = BaseVector::compareNulls(leftNull, rightNull, flags);
      if (result && *result == 0) {
        continue;
      }
      return result;
    }
```

Reply:
> /// Return std::nullopt if encountered in NullHandlingMode::StopAtNull mode.
  /// Support NullHandlingMode::NoStop mode as well and return
  /// std::optional<int32_t>(0) if null encountered.

This is not clear and may be wrong. I will update it as follows,
"Return std::nullopt if encountered in NullHandlingMode::StopAtNull mode. And return std::optional<int32_t>(0) if both sides are nulls in NullHandlingMode::NoStop mode."